ToolChain/Assembler/Parser/Parser.py

```diff
@@ -20,544 +19,368 @@
+            width = self.translateTextImmediateToImmediate(width)
+            instruction += bytes(0b0000) + bytes((width,)) + immediate.to_bytes(4, byteorder='big') + \
+                                                             immediate2.to_bytes(4, byteorder='big')
+
+        elif state == "STATE6":
+            width = arglist[0][1:-1]
+            immediate = self.translateTextImmediateToImmediate(arglist[1][1:])
+            register = self.translateRegisterNameToRegisterCode(arglist[2][1:])
+            width = self.translateTextImmediateToImmediate(width)
+            width = (width << 4) + register
+            instruction += bytes((width,)) + immediate.to_bytes(4, byteorder='big')
+
+        elif state == "STATE7":
+            width = arglist[0][1:-1]
+            register = self.translateRegisterNameToRegisterCode(arglist[1][1:])
+            immediate = self.translateTextImmediateToImmediate(arglist[2][1:])
+            width = self.translateTextImmediateToImmediate(width)
+            width = (width << 4) + register
+            instruction += bytes((width,)) + immediate.to_bytes(4, byteorder='big')
+
+        elif state == "STATE8":
+            width = arglist[0][1:-1]
+            register = self.translateRegisterNameToRegisterCode(arglist[1][1:])
+            register2 = self.translateRegisterNameToRegisterCode(arglist[2][1:])
+            width = self.translateTextImmediateToImmediate(width)
+            width = (width << 4) + register
+            instruction += bytes((width,)) + bytes((register2,))
+
+        elif state == "STATE9":
+            flag = self.translateTextFlagsToCodeFlags(arglist[0][1:-1])
+            if arglist[1][0] == "#":
+                immediate = self.translateTextImmediateToImmediate(arglist[1][1:])
+                instruction += bytes((flag,)) + bytes(0b0000) + immediate.to_bytes(4, byteorder='big')
+            elif ins in LABEL_INSTRUCTIONS:
+                instruction += bytes((flag,)) + bytes(0b0000) + b':' + arglist[1].encode("utf-8") + b':'
+
+        elif state == "STATE10":
+            flag = self.translateTextFlagsToCodeFlags(arglist[0][1:-1])
+            register = self.translateRegisterNameToRegisterCode(arglist[1][1:])
+            flag = (flag << 4) + register
+            instruction += bytes((flag,))
 
-        # Now, we need to put back the original memory reference if there was one
-        if memoryRefSymbol is not None:
-            binaryInstructionByteString = binaryInstructionByteString.replace(b"\xDE\xAD\xBE\xEF",
-                                                                              memoryRefSymbol.encode())
+        else:
+            # Nothing was valid in the first argument
+            raise ValueError("Invalid instruction format at line " + str(self.lineno))
 
-        return binaryInstructionByteString
+        return instruction
 
     def translateRegisterNameToRegisterCode(self, registerName: str=""):
```

Comment:
It took a bit of work since I had to go fix up some files in the capuaEnvironment folder etc., but this proved to be much easier to work with. It'll be in my next commit.